src/audio_to_image_model/audio_to_image_dataset_def.py

```diff
@@ -9,44 +10,68 @@
 # Audio Dataset
 def pad_or_truncate(mel_spectrogram, length):
     """
-    Pad or truncate the mel spectrogram to a fixed length
-    Args:
-        mel_spectrogram (torch.Tensor): The mel spectrogram tensor.
-        length (int): The fixed length to pad or truncate the mel spectrogram to
-    Returns:
-        torch.Tensor: The padded or truncated mel spectrogram.
-    """
+   Pad or truncate the mel spectrogram to a fixed length
+   Args:
+       mel_spectrogram (torch.Tensor): The mel spectrogram tensor.
+       length (int): The fixed length to pad or truncate the mel spectrogram to
+   Returns:
+       torch.Tensor: The padded or truncated mel spectrogram.
+   """
     mel_spectrogram = torch.squeeze(mel_spectrogram)
     if mel_spectrogram.shape[1] < length:
         padding = length - mel_spectrogram.shape[1]
         mel_spectrogram = torch.nn.functional.pad(mel_spectrogram, (0, padding), 'constant', 0)
     else:
         mel_spectrogram = mel_spectrogram[:, :length]
-
     mel_spectrogram = torch.unsqueeze(mel_spectrogram, 0)
     return mel_spectrogram
 
 
+def split_audio(mel_spectrogram_full, segment_length=5, sample_rate=32_000, n_fft=2048):
+    num_frames = mel_spectrogram_full.shape[2]
+    duration = num_frames * (n_fft // 2) / sample_rate  # Doc correct?
```

Comment:
# Doc correct? What do you mean with that?

Reply:
I do not know if the PyTorch documentation is up to date or not.